pkg/kubelet/kubeletconfig/util/codec/codec.go

```diff
@@ -125,3 +124,20 @@ func DecodeKubeletConfiguration(kubeletCodecs *serializer.CodecFactory, data []b
 
 	return internalKC, nil
 }
+
+// DecodeKubeletInstanceConfiguration decodes a serialized KubeletInstanceConfiguration to the external type,
+// use nil on defaultInstance to load the fields from YAML directly.
+func DecodeKubeletInstanceConfiguration(kubeletCodecs *serializer.CodecFactory, data []byte, defaultInstance *kubeletconfigv1beta1.KubeletInstanceConfiguration) (*kubeletconfigv1beta1.KubeletInstanceConfiguration, error) {
+	var (
+		obj runtime.Object
+		gvk *schema.GroupVersionKind
+	)
+
+	obj, gvk, _ = kubeletCodecs.UniversalDecoder().Decode(data, nil, defaultInstance)
```

Comment:
The Decode accepts an object to be decoded into, we need to use an external type for this to work, internal ones ended up losing the default value.